kola/tests/kubeadm/templates.go

```diff
@@ -0,0 +1,384 @@
+kind: Installation
+metadata:
+  name: default
+spec:
+  # Configures Calico networking.
+  calicoNetwork:
+    # Note: The ipPools section cannot be modified post-install.
+    ipPools:
+      - blockSize: 26
+        cidr: {{ .PodSubnet }}
+        encapsulation: VXLANCrossSubnet
+        natOutgoing: Enabled
+        nodeSelector: all()
+  flexVolumePath: /opt/libexec/kubernetes/kubelet-plugins/volume/exec/
+EOF
+
+{
+    systemctl enable --quiet --now kubelet
+    kubeadm config images pull
+    kubeadm init --config kubeadm-config.yaml
+    cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
+    cp -i /etc/kubernetes/admin.conf /home/core/.kube/config
+    chown -R core:core /home/core/.kube; chmod a+r /home/core/.kube/config;
+
+    kubectl create -f https://docs.projectcalico.org/manifests/tigera-operator.yaml
+    kubectl apply -f calico.yaml
+} > /dev/null 2>&1
+
+
+URL=$(kubectl config view -o jsonpath='{.clusters[0].cluster.server}')
+prefix="https://"
+short_url=${URL#"${prefix}"}
+token=$(kubeadm token create)
+certHashes=$(openssl x509 -pubkey -in /etc/kubernetes/pki/ca.crt | openssl rsa -pubin -outform der 2>/dev/null | openssl dgst -sha256 -hex | sed 's/^.* //')
+
+cat << EOF
+apiVersion: kubeadm.k8s.io/v1beta2
+kind: JoinConfiguration
+discovery:
+  bootstrapToken:
+    apiServerEndpoint: ${short_url}
+    token: ${token}
+    caCertHashes:
+    - sha256:${certHashes}
+controlPlane:
+nodeRegistration:
+  kubeletExtraArgs:
+    volume-plugin-dir: "/opt/libexec/kubernetes/kubelet-plugins/volume/exec/"
+EOF
+`
+
+	workerScript = `#!/bin/bash
+set -euo pipefail
+
+export RELEASE_VERSION={{ .ReleaseVersion }}
+export DOWNLOAD_DIR={{ .DownloadDir }}
+export PATH="${PATH}:${DOWNLOAD_DIR}"
+
+# create the required directory
+mkdir --parent \
+    /opt/cni/bin \
+    /etc/systemd/system/kubelet.service.d
+
+# we download and install the various requirements
+# * kubelet service and kubeadm dropin
+
+curl --retry-delay 1 \
+    --retry 60 \
+    --retry-connrefused \
+    --retry-max-time 60 \
+    --connect-timeout 20 \
+    --fail \
+    -sSL \
+    "https://raw.githubusercontent.com/kubernetes/release/${RELEASE_VERSION}/cmd/kubepkg/templates/latest/deb/kubelet/lib/systemd/system/kubelet.service" |
+    sed "s:/usr/bin:${DOWNLOAD_DIR}:g" |
+    tee /etc/systemd/system/kubelet.service
+
+curl --retry-delay 1 \
+    --retry 60 \
+    --retry-connrefused \
+    --retry-max-time 60 \
+    --connect-timeout 20 \
+    --fail \
+    -sSL \
+    "https://raw.githubusercontent.com/kubernetes/release/${RELEASE_VERSION}/cmd/kubepkg/templates/latest/deb/kubeadm/10-kubeadm.conf" |
+    sed "s:/usr/bin:${DOWNLOAD_DIR}:g" |
+    tee /etc/systemd/system/kubelet.service.d/10-kubeadm.conf
+
+systemctl enable --now kubelet
+
+cat << EOF > worker-config.yaml
+{{ .WorkerConfig }}
+EOF
+
+mainLink=$(ip route get 1 | grep -o -P ' dev .*? ' | cut -d ' ' -f 3)
+ipv4=$(ip -4 addr show "${mainLink}" | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
```

Comment:
```suggestion
systemctl start coreos-metadata
ipv4=$(cat /run/metadata/flatcar | grep -v IPV6 | grep IP | grep -E '(PRIVATE|LOCAL)' | cut -d = -f 2)
```
as in the existing K8s test